src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,114 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elements;
+    private int position;
+
+    public ArrayList() {
+        elements = new Object[DEFAULT_CAPACITY];
+        position = -1;
+    }
+
     @Override
     public void add(T value) {
-
+        if (isNeedToResize()) {
+            grow();
+        }
+        elements[++position] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndexToAdd(index);
+        if (isNeedToResize()) {
+            grow();
+        }
+        System.arraycopy(elements, index, elements, index + 1, size() - index);
+        elements[index] = value;
+        this.position++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndexToGet(index);
+        return (T) elements[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndexToGet(index);
+        elements[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndexToGet(index);
+        T toRemove = (T) elements[index];
+        if (!isLastIndex(index)) {
+            System.arraycopy(elements, index + 1, elements, index, position - index);
+        }
+        elements[position--] = null;
```

Comment:
I don't sure that we need assign it to null, we can only decrease size, so the next element addition will overwrite this value

Reply:
I think we could or could not - It really depends on your logic and approach.